client/my-sites/customer-home/cards/features/blogging-prompt/index.jsx

```diff
@@ -18,8 +20,24 @@ import './style.scss';
 const BloggingPromptCard = () => {
 	const dispatch = useDispatch();
 	const translate = useTranslate();
-	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const selectedSiteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	let sites = useSelector( ( state ) => getSitesItems( state ) );
+	let siteId = selectedSiteId;
+
+	// If no selected site ID, set site ID to first site with write site_intent
+	if ( siteId === null ) {
+		sites = sortBy( sites, [ 'ID' ] );
+		const blogs = Object.values( sites ).filter( ( site ) => {
+			return site.options?.site_intent === 'write';
+		} );
+		if ( blogs.length > 0 ) {
+			siteId = blogs[ 0 ].ID;
+		}
+	}
+
```

Comment:
What is the selected site when on reader? Is this just the most recent site viewed in the My Sites section?

I thinking falling back to the `getPrimarySiteId` selector if there are no blogs with write intent (which is a fairly recent thing added to onboarding) would be a good addition to this block.

Reply:
Good idea, I'll add a fallback to primary blog id if no blog with write intent is found.

In this case we need a site ID to allow us to get the list of prompts.

When rendering the prompts nav though the selected site ID is used in track stats - I'm not sure how to handle this.